Library/Homebrew/download_strategy.rb

```diff
@@ -390,7 +390,12 @@ def initialize(url, name, version, **meta)
     # Merge `:header` with `:headers`.
     if (header = meta.delete(:header))
       meta[:headers] ||= []
-      meta[:headers] << header
+
+      if header.is_a?(Array)
+        meta[:headers] += header
+      else
+        meta[:headers] << header
+      end
```

Comment:
What I'd probably do is make `cask/url.rb` use `meta[:headers]` instead and do `Array(header)` on the input to make it always an array.

Reply:
That sounds right, yes.

Reply:
@Bo98 I think I figured out what you were asking.

Let me know if I went down the wrong path. Otherwise, I think the PR should be ready.

Thanks!